### ISSUE TITLE
Bug: `no-restricted-globals` not checking types

### ISSUE THREAD

**Issue by T99** [typescript-eslint/typescript-eslint#11701]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&showAST=scope&code=PQKgBApgzgNglgOwC5gQewLQCdpK3AYyQgBMMBzGNAIwEMYoAuMAbQHIIss0s2AaMGwCqiJAEYAbAEEutAJ4BhNAih4ArkR79BAcQgJOtJDwBiahETjKlK9Zt4BdMCGABYAFAekcgA4QwACpiYAC8YCLIkjJY8jaqWBrGWADcXr7+AQBMoWB6BjFJZhZIVghxdknJQA&rules=N4XyA&tsConfig=N4XyA

### Repro Code

```TypeScript
/* eslint no-restricted-globals: ['error', 'Uint16ArrayConstructor', 'GeneratorFunctionConstructor'] */

type T1 = Uint16ArrayConstructor;
type T2 = GeneratorFunctionConstructor;
```

### Expected Result

I would expect that the usages of `Uint16ArrayConstructor` and `GeneratorFunctionConstructor` would be flagged by `no-restricted-globals` as an error.

### Actual Result

No error was reported.

### Additional Info

I have been attempting to configure ESLint for a project to report usages of the `React` global type as an error, in an effort to encourage the explicit importing of needed React symbols.

I had originally attempted to do so using `no-restricted-globals`, but it did not seem to be reporting usages of the `React` global in type declarations (i.e. `type MyType { field: React.MouseEvent<HTMLElement>; };`) as an error.

After a bit of searching, I encountered [this](673) bug that seems to be in line with my own experiences, but the bug was closed in 2022, citing that the rule should now report on types as well. The comment that closed out the bug included a link to a playground example that no longer seems to exemplify the functionality that I assume it was originally created to showcase.

I've included the playground link from that bug here as the example. I originally thought that it perhaps had something to do with the my local ESLint setup not being correctly configured to lint using type information, but the fact that the playground example also does not work leads me to believe this may not be the case.

**Comment by bradzacher** [comment]
https://typescript-eslint.io/rules/no-restricted-types

This rule is what you should be using to ignore types now. The landscape has changed since 2022 and core rules have changed since then.
